### PR TITLE
Bookstore jpa liquibase

### PR DESCRIPTION
**Pull request by azzilian**
Added - update and delete methods
UpdateBookRequestDto create for updating methods

### CODE REVIEW

File: src/main/java/com/onlinebookstore/onlinebookstore/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -38,8 +40,27 @@ public List<BookDto> findAll() {
     @Override
     public BookDto findById(Long id) {
         Book book = bookRepository.findById(id).orElseThrow(
-                () -> new EntityNotFoundException("Can't find book by id " + id)
+                () -> new DataProcessingException("Can't find book by id " + id)
```
Comment (karma-o) [inline src/main/java/com/onlinebookstore/onlinebookstore/service/impl/BookServiceImpl.java]: service layer should actually throw `EntityNotFoundException`, so let's revert this change, and check other places as well

File: src/main/java/com/onlinebookstore/onlinebookstore/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -38,8 +40,27 @@ public List<BookDto> findAll() {
...
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto update(Long id, UpdateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new DataProcessingException("Can't find book by id " + id)
+        );
```
Comment (karma-o) [inline src/main/java/com/onlinebookstore/onlinebookstore/service/impl/BookServiceImpl.java]: duplicating code across 3 methods

File: src/main/java/com/onlinebookstore/onlinebookstore/dto/UpdateBookRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.onlinebookstore.onlinebookstore.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class UpdateBookRequestDto {
```
Comment (karma-o) [inline src/main/java/com/onlinebookstore/onlinebookstore/dto/UpdateBookRequestDto.java]: no need to create another DTO when you have fields-wise identical CreateDTO, just rename `CreateBookRequestDto`  to `BookRequestDto`

File: src/main/java/com/onlinebookstore/onlinebookstore/model/Book.java [hunk 4]
```diff
@@ -9,10 +9,14 @@
...
 import lombok.Getter;
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Getter
 @Setter
+@SQLDelete(sql = "UPDATE Book SET isdeleted = true WHERE id=?")
+@Where(clause = "isdeleted = false")
```
Comment (karma-o) [inline src/main/java/com/onlinebookstore/onlinebookstore/model/Book.java]: `@Where` is deprecated, let's use `@SqlRestriction` instead, and if it won't work try changing the `is_deleted` type to `bit` in liquibase

File: src/main/java/com/onlinebookstore/onlinebookstore/controller/BookController.java [hunk 5]
```diff
@@ -33,4 +38,15 @@ public BookDto findById(@PathVariable Long id) {
     public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
+
+    @PutMapping("/{id}")
+    public BookDto updateBook(@PathVariable long id, @RequestBody UpdateBookRequestDto bookDto) {
+        return bookService.update(id, bookDto);
+    }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
```
Comment (karma-o) [inline src/main/java/com/onlinebookstore/onlinebookstore/controller/BookController.java:47]: add response status `CREATED` for `post` endpoint that you already have

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changeset/01-create-books-table.yml
```
Comment (karma-o) [inline src/main/resources/db/changelog/db.changelog-master.yaml]:
```suggestion
      file: db/changelog/changeset/01-create-books-table.yml

```

File: src/main/resources/db/changelog/changeset/01-create-books-table.yml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        - createTable:
+            tableName: books
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (karma-o) [inline src/main/resources/db/changelog/changeset/01-create-books-table.yml]: primary key is not null by default in MySQL, so we can remove that or leave it if you want to explicitly declare it

File: src/main/resources/db/changelog/changeset/01-create-books-table.yml [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+                  name: price
+                  type: numeric
+                  constraints:
+                    nullable: false
+              - column:
+                  name: description
+                  type: varchar(255)
+              - column:
+                  name: coverimage
+                  type: varchar(255)
```
Comment (karma-o) [inline src/main/resources/db/changelog/changeset/01-create-books-table.yml]: descriptions and URLs are usually pretty long bits of text, maybe it is worth raising their length to something like 500 at least?

**Review by karma-o — CHANGES_REQUESTED** [review]
Looks good so far, just take a look at the comments

File: src/main/resources/db/changelog/changeset/01-create-books-table.yml (hunk 7 above)
Comment (azzilian) [reply src/main/resources/db/changelog/changeset/01-create-books-table.yml]: A yes this is  reinsurance, I  think in this project I can remove it

**Review by andrii-hoienko — APPROVED** [review]
Good job!
